src/audio/eq_iir/eq_iir.c

```diff
@@ -699,236 +734,165 @@ static int eq_iir_verify_params(struct comp_dev *dev,
-		comp_cl_err(&comp_eq_iir, "eq_iir_func is not set");
-		return -EINVAL;
-	}
+	comp_info(dev, "eq_iir_set_config()");
 
-	return comp_set_state(dev, cmd);
+	return comp_data_blob_set(cd->model_handler, pos, data_offset_size, fragment,
+				  fragment_size);
 }
 
-static void eq_iir_process(struct comp_dev *dev, struct comp_buffer *source,
-			   struct comp_buffer *sink, int frames,
-			   uint32_t source_bytes, uint32_t sink_bytes)
+static int eq_iir_get_config(struct processing_module *mod,
+			     uint32_t config_id, uint32_t *data_offset_size,
+			     uint8_t *fragment, size_t fragment_size)
 {
-	struct comp_data *cd = comp_get_drvdata(dev);
+	struct sof_ipc_ctrl_data *cdata = (struct sof_ipc_ctrl_data *)fragment;
+	struct comp_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
 
-	buffer_stream_invalidate(source, source_bytes);
+	comp_info(dev, "eq_iir_get_config()");
 
-	cd->eq_iir_func(dev, &source->stream, &sink->stream, frames);
-
-	buffer_stream_writeback(sink, sink_bytes);
-
-	/* calc new free and available */
-	comp_update_buffer_consume(source, source_bytes);
-	comp_update_buffer_produce(sink, sink_bytes);
+	return comp_data_blob_get_cmd(cd->model_handler, cdata, fragment_size);
 }
 
-/* copy and process stream data from source to sink buffers */
-static int eq_iir_copy(struct comp_dev *dev)
+static int eq_iir_process(struct processing_module *mod,
+			  struct input_stream_buffer *input_buffers, int num_input_buffers,
+			  struct output_stream_buffer *output_buffers, int num_output_buffers)
 {
-	struct comp_copy_limits cl;
-	struct comp_data *cd = comp_get_drvdata(dev);
-	struct comp_buffer *sourceb;
-	struct comp_buffer *sinkb;
+	struct comp_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
 	int ret;
 
-	comp_dbg(dev, "eq_iir_copy()");
-
-	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer,
-				  sink_list);
-
 	/* Check for changed configuration */
 	if (comp_is_new_data_blob_available(cd->model_handler)) {
 		cd->config = comp_get_data_blob(cd->model_handler, NULL, NULL);
-		ret = eq_iir_setup(cd, sourceb->stream.channels);
+		ret = eq_iir_setup(mod, mod->stream_params->channels);
 		if (ret < 0) {
-			comp_err(dev, "eq_iir_copy(), failed IIR setup");
+			comp_err(dev, "eq_iir_process(), failed IIR setup");
 			return ret;
 		}
 	}
 
-	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
-				source_list);
+	cd->eq_iir_func(mod, &input_buffers[0], &output_buffers[0], input_buffers[0].size);
+	return 0;
+}
 
-	/* Get source, sink, number of frames etc. to process. */
-	comp_get_copy_limits_with_lock(sourceb, sinkb, &cl);
+/**
+ * \brief Set volume frames alignment limit.
+ * \param[in,out] source Structure pointer of source.
+ * \param[in,out] sink Structure pointer of sink.
+ */
+static void eqiir_set_alignment(struct audio_stream *source, struct audio_stream *sink)
+{
+#if XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4
```

Comment:
This HiFi3 version difference is not yet needed but I think it can be left this way since @andrula-song will implement a more efficient version of IIR in Q3.